packages/aws-cdk/test/api/hotswap/nested-stacks-hotswap.test.ts

```diff
@@ -0,0 +1,994 @@
+          Type: 'String',
+          Description: 'S3 bucket for asset',
+        },
+        S3BucketParam: {
+          Type: 'String',
+          Description: 'S3 bucket for asset',
+        },
+        S3KeyParam: {
+          Type: 'String',
+          Description: 'S3 bucket for asset',
+        },
+      },
+    },
+  });
+
+  setup.addTemplateToCloudFormationLookupMock(rootStack);
+  setup.addTemplateToCloudFormationLookupMock(testStack({
+    stackName: 'NestedStack',
+    template: {
+      Resources: {
+        Func: {
+          Type: 'AWS::Lambda::Function',
+          Properties: {
+            Code: {
+              S3Bucket: 'current-bucket',
+              S3Key: 'current-key',
+            },
+            FunctionName: 'my-function',
+          },
+          Metadata: {
+            'aws:asset:path': 'old-path',
+          },
+        },
+        GrandChildStack: {
+          Type: 'AWS::CloudFormation::Stack',
+          Properties: {
+            TemplateURL: 'https://www.magic-url.com',
+          },
+          Metadata: {
+            'aws:asset:path': 'one-lambda-stack-with-asset-parameters.nested.template.json',
+          },
+        },
+      },
+    },
+  }));
+  setup.addTemplateToCloudFormationLookupMock(testStack({
+    stackName: 'GrandChildStack',
+    template: {
+      Resources: {
+        Func: {
+          Type: 'AWS::Lambda::Function',
+          Properties: {
+            Code: {
+              S3Bucket: 'current-bucket',
+              S3Key: 'current-key',
+            },
+            FunctionName: 'my-function',
+          },
+          Metadata: {
+            'aws:asset:path': 'old-path',
+          },
+        },
+      },
+    },
+  }));
+
+
+  setup.pushNestedStackResourceSummaries('LambdaRoot',
+    setup.stackSummaryOf('NestedStack', 'AWS::CloudFormation::Stack',
+      'arn:aws:cloudformation:bermuda-triangle-1337:123456789012:stack/NestedStack/abcd',
+    ),
+  );
+
+  setup.pushNestedStackResourceSummaries('NestedStack',
+    setup.stackSummaryOf('GrandChildStack', 'AWS::CloudFormation::Stack',
+      'arn:aws:cloudformation:bermuda-triangle-1337:123456789012:stack/GrandChildStack/abcd',
+    ),
+  );
+  const cdkStackArtifact = testStack({ stackName: 'LambdaRoot', template: rootStack.template });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact, {
+    ChildS3BucketParam: 'child-bucket-param-value',
+    ChildS3KeyParam: 'child-key-param-value',
+    S3BucketParam: 'bucket-param-value',
+    S3KeyParam: 'key-param-value',
+  });
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(mockUpdateLambdaCode).toHaveBeenCalledWith({
+    FunctionName: 'my-function',
+    S3Bucket: 'bucket-param-value',
+    S3Key: 'key-param-value',
+  });
+  expect(mockUpdateLambdaCode).toHaveBeenCalledWith({
+    FunctionName: 'my-function',
+    S3Bucket: 'child-bucket-param-value',
+    S3Key: 'child-key-param-value',
+  });
+});
```

Comment:
Missing unit test that would catch the bug I mentioned in `EvaluateCloudFormationTemplate.createNestedEvaluateCloudFormationTemplate()`.